File: cpp-recitations-main/vectors/oddeven.cpp
#include <iostream>
#include <vector>
// https://www.geeksforgeeks.org/vector-in-cpp-stl/

int main() {

  int total_even = 0;
  int product_odd = 1;

  // For the below vector of values, write a program that displays the total
  // sum of the even elements and the product of all the odd elements.
  std::vector<int> vector = {2, 4, 3, 6, 1};


  std::cout << "Sum of even numbers is " << total_even << "\n";
  std::cout << "Product of odd numbers is " << product_odd << "\n";

  return 0;

}

File: py-recitations-solutions/vectors/oddeven.py
#oddeven.py

import random

# For Pythons "vectors" AKA lists or arrays (if you see through to using numpy),
# Write a program that displays the total sum of the even elements and the product of all the odd elements.

# Function(s) / Method(s)
def sumeven(v:list)->int:
    # Create a function to sum up the even numbers of a list/array
    s = 0
    for i in v:
        if i%2==0:
            s = i + s
    return s

def productodd(v:list)->int:
    # Create a function to multiply the odd numbers of a list/array
    p = 1
    for i in v:
        if i%2!=0:
            p = i*p
    return p

if __name__ == "__main__": 
    
    # list of integers
    vector=[]
    n=10
    for i in range(n):
        vector.append(random.randint(1,9))
    
    # vector = [2, 4, 3, 6, 5]

    # Sum of Even elements 
    se = sumeven(vector)

    # Product of Odd elements 
    po = productodd(vector)

    # Output the result to the user.
    print(f"List : {vector}\nSum of even : {se}\nProduct of odd : {po}\n")